src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -56,6 +56,16 @@ public class BlockStoreBehavior : NetworkPeerBehavior, IBlockStoreBehavior
 
         private bool preferHeaderAndIDs;
 
+        /// <summary>Hash of the last block we've sent to the peer in response to "getblocks" message,
+        /// or <c>null</c> if the peer haven't used "getblocks" message or if we sent a tip to it already.</summary>
+        /// <remarks>
+        /// In case the peer is syncing using outdated "getblocks" message, we need to maintain 
+        /// the hash of the last block we sent to it in an inventory batch. Once the peer asks 
+        /// for block data of the block with this hash, we will send a continuation inventory message.
+        /// This will cause the peer to ask for more.
+        /// </remarks>
+        private uint256 getBlocksBatchLastItemHash;
```

Comment:
Shall we add a todo that this needs to go away once we fork away from X?

Reply:
it does not really hurt to support this thing, but it will become really obsolete then ... that's a long shot issue to have, but let me add such a todo comment just above this line

```
            this.CanRespondToGetBlocksPayload = true;

```

Reply:
oh well it was there already:

```
                case GetBlocksPayload getBlocksPayload:
                    // TODO: this is not used in core anymore consider deleting it
                    // However, this is required for StratisX to be able to sync from us.
```

Reply:
so no change

Reply:
Ah cool ok I missed it.